src/ngx_pagespeed.cc

```diff
@@ -2967,10 +2979,18 @@ ngx_int_t ps_init_module(ngx_cycle_t* cycle) {
   } else {
     delete cfg_m->driver_factory;
     cfg_m->driver_factory = NULL;
+    active_driver_factory = NULL;
   }
   return NGX_OK;
 }
 
+void ps_exit_child_process(ngx_cycle_t* cycle) {
+  ps_main_conf_t* cfg_m = static_cast<ps_main_conf_t*>(
+      ngx_http_cycle_get_module_main_conf(cycle, ngx_pagespeed));
+  NgxBaseFetch::Terminate();
+  cfg_m->driver_factory->ShutDown();
```

Comment:
@jeffkaufman 
I do think we need to remove the DCHECK in SystemCaches::ShutDown(): I need to explicitly call ShutDown() on the driver factory here and destruct it at another point in time (pool cleanup handler). When NgxBaseFetch::Terminate() isn't able to round up within its timeout period, it should shutdown the driver factory as soon as possible.